particles/Tutorial/Kotlin/Blackjack/DealingShoe.kt

```diff
@@ -0,0 +1,59 @@
+package arcs.tutorials.blackjack
+
+import kotlin.random.Random
+import arcs.sdk.Handle
+
+import arcs.sdk.Utils.log
+
+class DealingShoe : AbstractDealingShoe() {
+    val cardPresent = "x"
+    val cardAbsent = "-"
+    val numDecks = 1
+    val totalCards = numDecks * 52
+    val emptyDeck = cardAbsent.repeat(totalCards)
+
+    override fun getTemplate(slotName: String) = """
+        Card is <span>{{nextCard}}</span>
+     """.trimIndent()
+
+    override fun populateModel(slotName: String, model: Map<String, Any>): Map<String, Any> {
+        return model + mapOf("nextCard" to nextCard.toString())
+    }
+
+    override fun onHandleUpdate(handle: Handle) {
+        if (handle.name != "cardRequest") { return }
+        val request = cardRequest.get()
+        if (request == null) { return }
+        val card: Card? = pickACard()
+        if (card == null) { return }
+        nextCard.set(DealingShoe_NextCard(
+                player = request.player,
+                card = card.value.toDouble()))
+        this.renderOutput()
+    }
+
+    /**
+     * Returns an initialized set of decks.
+     */
+    fun initializedDecks(): DealingShoe_Decks{
+        var d = DealingShoe_Decks()
+        d.cards = cardPresent.repeat(totalCards)
+        return d
+    }
+
+    fun pickACard(): Card? {
+        var localDecks = decks.get() ?: initializedDecks()
+        var choice = Random.nextInt(totalCards)
+        var cards = localDecks.cards
+        if (cards.equals(emptyDeck)) return null
+        // This could be done more efficiently, but should suffice for now.
+        var readCards = 0
+        while (readCards < totalCards && cards[choice] == cardAbsent[0]) {
+            choice = (choice + 1) % totalCards
+            ++readCards
+        }
+        localDecks.cards = cards.replaceRange(choice, choice + 1, cardAbsent)
+        decks.set(localDecks)
+        return Card(choice % 52)
```

Comment:
Would this be simpler if you had a local list that contained all the cards "left" in the deck. So it would start with 0, 1, 2, ... 51. As cards were dealt they could be removed from the list. Then you would just need to get a random card from the deck, remove it from the local list, and return the chosen card.

Reply:
I complicated this by making `decks` an entity. If `decks` was local to the class, I would do it like this. Let me know what you think of the comment about `decks` above and I will change this accordingly.

Reply:
> I wondered the same thing too. I looked for a place for Kotlin particles under `particles/`, but did not find one. That is why I left it here. Happy to move the code to wherever you think is appropriate.

I don't think we have a place under particles/ yet, so the high level should be fine.

Reply:
WIll do as a last step before merging the PR.